diff -ruN libcaldav-0.6.2/src/caldav.c libcaldav-0.6.2a/src/caldav.c
--- libcaldav-0.6.2/src/caldav.c	2010-09-11 08:10:15.000000000 -0700
+++ libcaldav-0.6.2a/src/caldav.c	2010-11-02 21:53:34.000000000 -0700
@@ -35,16 +35,8 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#include <gnutls/gnutls.h>
-#include <gcrypt.h>
-#include <pthread.h>
-#include <errno.h>
-
-GCRY_THREAD_OPTION_PTHREAD_IMPL;
 
 static void init_runtime(runtime_info* info) {
-	gcry_control (GCRYCTL_SET_THREAD_CBS, &gcry_threads_pthread);
-    gnutls_global_init();
     if (! info)
 		return;
     if (! info->error)

diff -ruN libcaldav-0.6.2/src/options-caldav-server.c libcaldav-0.6.2a/src/options-caldav-server.c
--- libcaldav-0.6.2/src/options-caldav-server.c	2010-11-02 21:56:33.000000000 -0700
+++ libcaldav-0.6.2a/src/options-caldav-server.c	2010-09-11 04:36:42.000000000 -0700
@@ -87,8 +87,7 @@
 				parse_response(CALDAV_OPTIONS, code, chunk.memory)) {
 			enabled = TRUE;
 			if (! test) {
-				result->msg = g_strdup(
-						get_response_header("Allow", headers.memory, FALSE));
+				result->msg = get_response_header("Allow", headers.memory, FALSE);
 			}
 		}
 		else {
